README.md

```diff
@@ -1,2 +1,263 @@
+A helper whose return value is a function that calls the supplied callback with its argument and returns that argument.
+Often used for performing side effects within a functional pipeline
+
+For example:
+```js
+import {flow} from 'lodash/fp'
+
+const repeatTitle = flow(
+  title => `${title} ${title}`,
+  tap(repeatedTitle => {
+    document.title = repeatedTitle
+  }),
+  repeatedTitle => `${repeatedTitle}!`
+)
+
+repeatTitle("bye") // returns "bye bye!" and sets document title to "bye bye"
+```
+
+`tap()` is a standard functional helper, see the [Ruby](https://apidock.com/ruby/Object/tap) and [Ramda](https://ramdajs.com/0.21.0/docs/#tap) versions
+
+### `log()`
+
+```js
+log(
+  loggingKey: string
+): Function
+```
+
+A helper whose return value is a function that logs its argument to the console under the given key and returns that argument.
+Useful for debugging functional pipelines
+
+For example:
+```js
+import {flow} from 'lodash/fp'
+
+const complicatedPipeline = flow(
+  log('initial'),
+  val => val * 2,
+  log('doubled'),
+  doubledVal => doubledVal + 4,
+  log('added'),
+  addedVal => addedVal ? 'yes' : 'no'
+)
+
+complicatedPipeline(-2) // returns "no" and logs the equivalent of the following to the console:
+// console.log({initial: -2})
+// console.log({doubled: -4})
+// console.log({added: 0})
+```
+
+### `reversed()`
+
+```js
+reversed(
+  array: Array<any>
+): Function
+```
+
+Returns a new array which is the reversed version of its argument. Doesn't mutate its argument
+
+For example:
+```js
+reversed([1, 2, 3]) // returns [3, 2, 1]
+```
+
+### `ensureArray()`
+
+```js
+ensureArray(
+  arrayOrScalar: any
+): Array<any>
+```
+
+If passed an array, returns that array. If passed anything else, returns a single-element array whose only element is that argument
+
+For example:
+```js
+ensureArray([1, 2, 3]) // returns [1, 2, 3]
+
+ensureArray('abc') // returns ['abc']
+```
+
+### `callWith()`
+
+```js
+callWith(
+  argument: any
+  ...
+): Function
+```
+
+Returns a function which expects to get passed a function and applies the original arguments to that function
+
+For example:
+```js
+callWith(1, 2)((a, b) => a + b) // returns 3
+```
+
+### `removeProps()`
+
+```js
+removeProps(
+  propNameOrArrayOfPropNames: string | Array<string>
+): Function
+```
+
+Returns a function which expects an object and returns a new copy of that object without the properties under the supplied keys
+
+For example:
+```js
+removeProps('a')({a: 1, b: 2}) // returns {b: 2}
+
+removeProps(['a', 'b'])({a: 1, b: 2, c: 3}) // returns {c: 3}
+```
+
+### `mapToObject()`
+
+```js
+mapToObject(
+  callback: (value: any, key: string | number) => [key: string, value: any]
+): Function
+```
+
+Returns a function which "maps" the supplied callback to every element in its argument and builds a single object as its return value. The callback should return a two-element `[key, value]` array in the style of [`fromPairs()`](https://lodash.com/docs/4.17.11#fromPairs)
```

Comment:
I'd specify what happens if the callback produces duplicate keys.

I wouldn't consider something strictly a "map" if the number of elements in the output can be different than the number of elements in the input (which you can do here by returning duplicate keys). Maybe `transformObject` instead?

Reply:
I feel much less strongly about this one than the others. I wouldn't find it weird to "map" as Set, which would have the same issue where the number of elements can change. 🤷‍♂

Reply:
I think it's a good point though, and if I'm roughly going for Lodash-consistent naming, there `map()` always returns an array

What do you think of eg `transformToObject()` vs `reduceToObject()`? My initial reaction is that when I see `reduce` in the name I expect to call it using `reduce()`-style accumulator/callbacks (which you don't here)

That's probably why I called it a `map` is that when using it, it feels like a `map` since you're supplying a callback that gets invoked for each element

Reply:
Yeah, I like `transformToObject` better than `reduceToObject` for the reason you give. Something else that might make naming a bit easier would be to namespace these functions by their input type, so you could call `FPObject.transform`, or something like that.